geotf/src/geodetic_converter.cc

```diff
@@ -0,0 +1,441 @@
+#include <geotf/geodetic_converter.h>
+namespace geotf {
+// Initialize frame definitions from rosparams
+void GeodeticConverter::initFromRosParam(const std::string& prefix) {
+  GeodeticConverter converter;
+  ros::NodeHandle nh;
+  if (!nh.ok() || !nh.hasParam(prefix)) {
+    ROS_WARN("[GeoTF] No GeodeticTF Transformations found.");
+    return;
+  }
+  XmlRpc::XmlRpcValue yaml_raw_data;
+  nh.getParam("/geotf", yaml_raw_data);
+
+  XmlRpc::XmlRpcValue& frame_definitions = yaml_raw_data["Frames"];
+  for (auto it = frame_definitions.begin();
+       it != frame_definitions.end(); ++it) {
+
+    const std::string frame_name = it->first;
+    XmlRpc::XmlRpcValue& xmlnode = it->second;
+
+    std::string frame_type = xmlnode["Type"];
```

Comment:
```suggestion
    auto frame_type = xmlnode["Type"];
```

Reply:
I think here std::string could even be needed for template deduction stuff - so I leave it.